system/jlib/jfcmp.hpp

```diff
@@ -43,9 +43,9 @@ class jlib_decl CFcmpCompressor : public CSimpleInterfaceOf<ICompressor>
     virtual void setinmax() = 0;
     virtual void flushcommitted() = 0;
 
-    void initCommon()
+    void initCommon(size32_t initialSize)
     {
-        blksz = inma.capacity();
+        blksz = initialSize;
```

Comment:
capacity() is max capacity. If the compressor has previously been opened with a larger initial size, then that's what inma.capacity() will be. i.e. the code doesn't shrink the buffer each open.

So initialSize is the size passed into open() and will be used for blksz.